src/components/CreateNewPin.js

```diff
@@ -8,25 +13,178 @@ import {
   CreatePinButton,
   UploadNewPinImageLabel,
   UploadNewPinImageInput,
+  PreviewImage,
 } from "../styles/CreateNewPin.module";
+import imageCompression from "browser-image-compression";
+
+const firebaseConfig = {
+  // eslint-disable-next-line no-undef
+  apiKey: process.env.REACT_APP_FIREBASE_APIKEY,
+  authDomain: process.env.REACT_APP_FIREBASE_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECTID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGEBUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGINGSENDERID,
+  appId: process.env.REACT_APP_FIREBASE_APPID,
+  measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENTID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const db = getFirestore(app);
 
 function CreateNewPin() {
-  const [pinName, setPinName] = useState();
-  const [pinDescription, setPinDescription] = useState();
-  const [pinLink, setPinLink] = useState();
+  const [pinName, setPinName] = useState("");
+  const [pinDescription, setPinDescription] = useState("");
+  const [pinLink, setPinLink] = useState("");
+  const [pinImage, setPinImage] = useState();
+  const [isGetPinImage, setIsGetPinImage] = useState(false);
+  const [selectedFile, setSelectedFile] = useState();
+  const [preview, setPreview] = useState();
+  const [isPinCreated, setIsPinCreated] = useState(false);
+
+  const redirect = useNavigate();
+
+  async function handleImageUpload(e) {
+    const imageFile = e.target.files[0];
+
+    // show image preview
+    if (!e.target.files || e.target.files.length === 0) {
+      setSelectedFile(undefined);
+
+      return;
+    }
+    setSelectedFile(imageFile);
+
+    // compress image
+    console.log(`original size ${imageFile.size / 1024 / 1024} MB`);
+    const imageName = imageFile.name.split(".");
+    localStorage.setItem("uploadedImageName", imageName[0]);
+
+    const options = {
+      maxSizeMB: 1,
+      maxWidthOrHeight: 1920,
+      useWebWorker: true,
+    };
+    try {
+      const compressedFile = await imageCompression(imageFile, options);
```

Comment:
Just fyi, you could compress images on the front-end by using canvas as well.

Reply:
got it! thanks for the info~